mach_nix/deptree.py

```diff
@@ -45,13 +45,17 @@ def remove_circles_and_print(pkgs: Iterable[ResolvedPkg], nixpkgs: NixpkgsIndex)
     indexed_pkgs = {p.name: p for p in sorted(pkgs, key=lambda p: p.name)}
     roots: Iterable[ResolvedPkg] = sorted([p for p in pkgs if p.is_root], key=lambda p: p.name)
 
+    def get_build_inputs(pkg: ResolvedPkg):
```

Comment:
This was just the simplest possible abstraction of existing code that was already duplicating this logic in a few places. Some of the PRs open by @tomprince have a much more thorough reworking of this file, so I tried to only touch it the minimum required to fix the NoneType + NoneType error